third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -881,6 +881,20 @@ func resourceContainerCluster() *schema.Resource {
 											Optional: true,
 											Default:  false,
 									},
+									"enable_resource_consumption_metering": {
```

Comment:
Can you break this schema out so that it's just the following? As well as make the required transformations in code to send/read the right request. [`enable_shielded_nodes`](https://www.terraform.io/docs/providers/google/r/container_cluster.html#enable_shielded_nodes) is an example where we made a similar transformation, I believe.

```
"enable_resource_consumption_metering": {
    Type:     schema.TypeBool,
    Optional: true,
    Default:  true,
},
```

(defaulting to whatever the API does)

Reply:
OK, I'm trying to find the time for this - I'm having a hard time figuring how to nest the Shielded style code into another resource block. 
I do think more fields are coming to this block in the future, in case that matters.

Reply:
I'm not aware of any additional subfields upcoming, so I'd rather we pull it up. We can add them in a compatible way if any are added. 

https://github.com/terraform-providers/terraform-provider-google-beta/commit/6e56d74775fc4a8d75aaed5a944eafb4ebf16b03 is a good reference for what adding `enable_shielded_nodes` looked like (although it's munged with shielded_instance_config. Only the parts in google_container_cluster.go matter.)